official/resnet/keras/keras_imagenet_main.py

```diff
@@ -0,0 +1,237 @@
+                                            imagenet_main._DEFAULT_IMAGE_SIZE,
+                                            imagenet_main._DEFAULT_IMAGE_SIZE,
+                                            imagenet_main._NUM_CHANNELS,
+                                            imagenet_main._NUM_CLASSES,
+                                            dtype=flags_core.get_tf_dtype(
+                                                flags_obj))
+
+  else:
+    train_input_dataset = imagenet_main.input_fn(
+        True,
+        flags_obj.data_dir,
+        batch_size=per_device_batch_size,
+        num_epochs=flags_obj.train_epochs,
+        num_gpus=flags_obj.num_gpus,
+        parse_record_fn=imagenet_main.parse_record)
+
+    eval_input_dataset = imagenet_main.input_fn(
+        False,
+        flags_obj.data_dir,
+        batch_size=per_device_batch_size,
+        num_epochs=flags_obj.train_epochs,
+        num_gpus=flags_obj.num_gpus,
+        parse_record_fn=imagenet_main.parse_record)
+
+  # Set environment vars and session config
+  session_config = resnet_run_loop.set_environment_vars(flags_obj)
+  session = tf.Session(config=session_config)
+  tf.keras.backend.set_session(session)
+
+  # Use Keras ResNet50 applications model and native keras APIs
+  # initialize RMSprop optimizer
+  # TODO(anjalisridhar): Move to using MomentumOptimizer.
+  opt = tf.train.GradientDescentOptimizer(learning_rate=0.0001)
+
+  strategy = distribution_utils.get_distribution_strategy(
+      num_gpus=flags_obj.num_gpus)
+
+  model = resnet_model.ResNet50(classes=imagenet_main._NUM_CLASSES,
+                                weights=None)
+
+  # Hardcode learning phase to improve perf by getting rid of conds
+  # in the graph. You can do this if you are either only training or evaluating.
+  if flags_obj.train_only:
+    tf.keras.backend.set_learning_phase(True)
+
+  if flags_obj.eval_only:
+    tf.keras.backend.set_learning_phase(False)    
+
+  model.compile(loss=softmax_crossentropy_with_logits,
+                optimizer=opt,
+                metrics=['accuracy'],
+                distribute=strategy)
+  time_callback = TimeHistory(flags_obj.batch_size)
+
+  steps_per_epoch = imagenet_main._NUM_IMAGES['train'] // flags_obj.batch_size
+  model.fit(train_input_dataset,
+            epochs=flags_obj.train_epochs,
+            steps_per_epoch=steps_per_epoch,
+            callbacks=[time_callback],
+            verbose=0)
+
+  num_eval_steps = imagenet_main._NUM_IMAGES['validation'] // flags_obj.batch_size
+  eval_output = model.evaluate(eval_input_dataset,
+                               steps=num_eval_steps,
+                               verbose=0)
+  print('Test loss:', eval_output[0])
+
+  # If you have set FLAGS.train_epochs to be 1 then we cannot calculate
+  # samples/s in a meaningful way since the first epoch takes the longest.
+  if flags_obj.train_epochs == 1:
+    print('Please increase the number of train_epochs if you want to '
```

Comment:
The comment and the printed message are kinda saying the same thing. Let's get rid of the comment and explain in the user-visible message:

```python
print('samples/sec is not meaningful when --train_epochs=1 since the first epoch takes the longest. Increase --train_epochs to see samples/s estimates')
```

Or maybe, better still - can we just discard the first batch instead of discarding the whole first epoch? For example, by subtracting the time for the first batch from the total epoch time.

Also, how about moving all this time calculation into the `TimeHistory` hook as a method:

```python
def images_per_second(self): ...
```

That makes it easier to see the purpose of all the timestamps collected as the collection and use is encapsulated in one place.

Reply:
Removed this code block since I am already printing out images_per_second as part of the callback. We don't need this additional calculation.